app/controllers/hubstats/deploys_controller.rb

```diff
@@ -30,11 +30,19 @@ def show
     end
 
     def create
-      @deploy = Deploy.new(params[:deploy])
-      if @deploy.save
-        redirect_to :action => 'index'
-      else
+      if params[:deployed_by].nil? || params[:git_revision].nil? || params[:repo_name].nil?
         render :nothing => true, :status => 400
+      else
+        @deploy = Deploy.new
+        @deploy.deployed_at = params[:deployed_at]
+        @deploy.deployed_by = params[:deployed_by]
+        @deploy.git_revision = params[:git_revision]
+        @deploy.repo_id = Hubstats::Repo.where(full_name: params[:repo_name]).first.id.to_i
```

Comment:
We don't want to 500.  Let's return a 400 for this case as well.  As we now have multiple cases that result in a 400 Bad Request response, let's try to add a error message in the response body with a short statement about why its bad. You will have to make the message for your first two cases here, but if the save fails the model validation should make errors available for your to use.